### DISCUSSION TITLE
Pushing a button to trigger an update?

### DISCUSSION THREAD

**Discussion by jouellnyc** [micropython/micropython#10551] [Hardware & Peripherals] [answered]
Howdy. Coming back to the ESP32-S3-BOX LITE. Such a great little unit. I am trying to implement a 'update' button whereby a user can push one of the 3 buttons on the front and the device would use `ugit` to pull an update and reboot -- that sort of thing.  

I have 2 questions. 

1) Anyone know the GPIO numbers for the buttons? 

I looked at the PDFs/docs and the schematics but was not able to figure that out.

I also tried pushing a button while looping through:

```
>>> for x in range(1,100):
    try:
        button = machine.Pin(x, machine.Pin.IN, machine.Pin.PULL_UP)
        print(f"Button state {x}", button.value())
        import time
        time.sleep(0.1)
    except Exception:
        print(f" invalid pin: {x}")

```
to see if I could reveal the location. Hopefully it's actually available to micropython?

2) Is this very doable or even wise? 

My device grabs data from a server and displays it on a screen, then sleeps. 
Would I be better suited to look into using ISR? (The docs say  an ISR should be short and that's not exactly a short series of actions)

Thanks much in advance

**Comment by robert-hh** [comment, marked as answer]
As far as I can tell from the schematics at https://github.com/espressif/esp-box, they are connected all together for a button_adc input, with different resistors to GND. So you can tell by the value read form ADC which button is pressed, The adc input is GPIO1, the pull-up resistor is 10k, downlink resistors at the buttons are 3.3k, 15k and 27k.

**Reply by jouellnyc** [reply]
Thanks very much @robert-hh for the quick reply!

Funny, I did  see that at https://github.com/espressif/esp-box/blob/master/hardware/esp32_s3_box_lite_Button_V1.1/schematic/SCH_ESP32-S3-BOX-Lite_Button_V1.1_20211125.pdf but my electronics foo is not strong enough at the moment to put that all together in a coherent sentence like you just did. (I naively thought each button would be it's own gpio pin). So that's great. Thank you! 

I also did see the Buttons/ADC part in 
https://github.com/espressif/esp-box/blob/master/hardware/esp32_s3_box_lite_MB_V1.1/schematic/SCH_ESP32-S3-BOX-Lite_MB_V1.1_20211221.pdf, but somehow did not see the GPIO1 text in red text at all :(. Probably since you highlighted it, I can now 'see' it.  

So it looks like I have a bit of homework to and have a some good  next steps. 

Thanks ever so much!

**Reply by jouellnyc** [reply]
I'm not sure how valuable this is but, for someone doing the same as me:

If I touch the 3rd button (last on the right), the readings change and can be gotten:

`
from machine import Pin, SPI, ADC
adc = ADC(Pin(1))
`

#Unpushed
>>> adc.read_u16()
65535

#Pushed
>>> adc.read_u16()
55261

So I can do this:

```
def check_button3():
    if str(adc.read_u16()).startswith('55'):
        return True

```
I tested this probably 15-20 times. This works for me and my project.

Thanks again @robert-hh
